src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
-package core.basesyntax;
-
-public class MyHashMap<K, V> implements MyMap<K, V> {
-
-    @Override
-    public void put(K key, V value) {
-
-    }
-
-    @Override
-    public V getValue(K key) {
-        return null;
-    }
-
-    @Override
-    public int getSize() {
-        return 0;
-    }
-}
+package core.basesyntax;
+
+import java.util.Objects;
+
+public class MyHashMap<K, V> implements MyMap<K, V> {
+
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private int capacity;
+    private final float loadFactor;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int capacity, float loadFactor) {
+        this.capacity = capacity;
+        this.loadFactor = loadFactor;
+        this.table = new Node[capacity];
+    }
+
+    @Override
+    public int getSize() {
+        return this.size;
+    }
+
+    public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+        }

```

Comment:
Special handling of null key is redundant. Your getIndex method handles it properly. Objects.equals() also works with null.
So null key may be treated as any other key in map